### PR TITLE
Refactor/ast transformers part 1

### PR DESCRIPTION
**Pull request by mathpirate**
Branch off part 1 of the AST transformers refactor; goal is to land core refactor including many correctness fixes; proper handling of commontools_1 symbol resolution issue is being moved to a follow-on PR.

This PR will need to implement a TBD quick fix for that issue in order to land.
    

---

## Summary by cubic
Refactors the TypeScript AST transformers into a new package (@commontools/ts-transformers) with a modular OpaqueRef rewrite and schema injection, and moves fixture testing to a shared test-support package. This cleans up js-runtime, improves correctness (JSX/element access/conditionals), and lays groundwork for CT-891.

- **Refactors**
  - New @commontools/ts-transformers package exporting:
    - Modular OpaqueRef transformer (JSX rule, schema injection rule).
    - Shared transformation context and import manager.
  - js-runtime now consumes transformers from @commontools/ts-transformers; legacy transformer files removed.
  - New @commontools/test-support for reusable fixture runner; fixtures moved from js-runtime into ts-transformers and schema-generator.
  - deno.json updates to include new packages and fixture paths; tests support UPDATE_GOLDENS.
  - Type env now includes commontools.d.ts.

- **Bug Fixes**
  - More accurate reactive rewrites:
    - Handles element and property access, binary ops, template strings, conditionals, and nested cases with minimal wrapping.
    - Safer event handler detection to avoid over-wrapping.
  - Recipe schema type arguments injected only when present; schema transform is skipped without the directive (covered by a new compiler test).
  - Temporary workaround for the commontools_1 alias to unblock AMD output; a robust symbol-resolution fix will follow.

### CODE REVIEW

**Review by jsantell — COMMENTED** [review]
Looking great!


Some other scenarios that could occur that may not yet be addressed (noting for future, we could require specific authored code in the mean time):

## Author globs commontools

```tsx
import * as ct from "commontools"
```

This should still be valid, and the transforms would apply `ct.derive`

## Author renames imports

```tsx
import { h, derive as ctDerive } from "commontools"
```

This should be valid, and the transforms apply `ctDerive`. This would push us towards referencing these "builtins" by symbol rather than e.g. by string.

## Author does not provide *any* imports from commontools

This could occur in a shared types/schema file for example

```tsx
export type MySchema = { name: string };
```

IIUC, we can't inject a fresh import here because we're already past the Binding stage in TSC. Without changing the main transformation pipeline (maybe we want to explore 2-pass options in the future), we could perform some preprocessing on the source file strings when 1) cts-enabled, and 2) naive regex matching to determine that commontools has *not* been imported. In that case, we could inject a `import * as __ct from "commontools"` or something, if we support glob style imports

**Comment by mathpirate** [comment]
> Looking great!
> 
> Some other scenarios that could occur that may not yet be addressed (noting for future, we could require specific authored code in the mean time):
> 
> ## Author globs commontools
> ```tsx
> import * as ct from "commontools"
> ```
> 
> This should still be valid, and the transforms would apply `ct.derive`
> 
> ## Author renames imports
> ```tsx
> import { h, derive as ctDerive } from "commontools"
> ```
> 
> This should be valid, and the transforms apply `ctDerive`. This would push us towards referencing these "builtins" by symbol rather than e.g. by string.
> 
> ## Author does not provide _any_ imports from commontools
> This could occur in a shared types/schema file for example
> 
> ```tsx
> export type MySchema = { name: string };
> ```
> 
> IIUC, we can't inject a fresh import here because we're already past the Binding stage in TSC. Without changing the main transformation pipeline (maybe we want to explore 2-pass options in the future), we could perform some preprocessing on the source file strings when 1) cts-enabled, and 2) naive regex matching to determine that commontools has _not_ been imported. In that case, we could inject a `import * as __ct from "commontools"` or something, if we support glob style imports

Thank you! I added this in the transformers_notes.md file as a set of future tasks.

File: packages/ts-transformers/src/core/common-tools-symbols.ts [hunk 1]
```diff
@@ -0,0 +1,185 @@
+import ts from "typescript";
+
+// Constants for identifying CommonTools declarations
+const COMMONTOOLS_FILENAME_FRAGMENT = "/commontools/";
```
[the remark this answers is not in the document]
Comment (mathpirate) [reply packages/ts-transformers/src/core/common-tools-symbols.ts]: thanks, i removed this entirely and everything still works :)

File: packages/ts-transformers/src/opaque-ref/call-kind.ts [hunk 2]
```diff
@@ -0,0 +1,279 @@
...
+  "Array",
+  "ReadonlyArray",
+]);
+
+const OPAQUE_REF_OWNER_NAMES = new Set([
+  "OpaqueRefMethods",
+  "OpaqueRef",
+]);
+
+const COMMONTOOLS_PATH_FRAGMENT = "commontools";
```
[the remark this answers is not in the document]
Comment (mathpirate) [reply packages/ts-transformers/src/opaque-ref/call-kind.ts]: yeah good point, i changed this to use the constant defined in common-tools-symbols that is the .d.ts file

File: packages/js-runtime/utils.ts [hunk 3]
```diff
@@ -15,11 +16,13 @@ export const getTypeScriptEnvironmentTypes = (() => {
     const es2023 = await cache.getText("types/es2023.d.ts");
     const jsx = await cache.getText("types/jsx.d.ts");
     const dom = await cache.getText("types/dom.d.ts");
+    const commontools = await cache.getText("types/commontools.d.ts");
```
[the remark this answers is not in the document]
Comment (mathpirate) [reply packages/js-runtime/utils.ts]: ok cool, i removed this and everything still works, thank you!

File: packages/js-runtime/deno.json [hunk 4]
```diff
@@ -1,11 +1,12 @@
...
   "name": "@commontools/js-runtime",
   "tasks": {
-    "test": "deno test --allow-read --allow-write --allow-run --allow-env=API_URL,\"TSC_*\",NODE_INSPECTOR_IPC,VSCODE_INSPECTOR_OPTIONS,NODE_ENV test/*.test.ts"
+    "test": "deno test --allow-read --allow-write --allow-run --allow-env=UPDATE_GOLDENS,API_URL,\"TSC_*\",NODE_INSPECTOR_IPC,VSCODE_INSPECTOR_OPTIONS,NODE_ENV test/*.test.ts"
   },
   "imports": {
     "typescript": "npm:typescript",
-    "source-map-js": "npm:source-map-js@^1.2.1"
+    "source-map-js": "npm:source-map-js@^1.2.1",
+    "@commontools/ts-transformers": "../ts-transformers/src/mod.ts"
```
[the remark this answers is not in the document]
Comment (mathpirate) [reply packages/js-runtime/deno.json]: thanks, removed

**Comment by mathpirate** [comment]
New modular transformer stack.

  - Stand up @commontools/ts-transformers: normalized data-flow analysis, rule-based opaque-ref rewrite, schema-injection rule, import helpers, and comprehensive fixture coverage
  - Share transformer utilities across the workspace (common-tools-imports, common-tools-symbols, call-kind detection, binding plans) with robust helper reuse and canonical derive substitution
  - Wire the new package into @commontools/js-runtime: use the modular transformers, tighten virtual FS/module resolution, streamline environment lib handling, and keep compiler tests up to date with explicit CommonTools typings
  - Document roadmap and open gaps (transformers_notes.md), expand fixtures/tests to guard edge cases, and align schema generation via schema-generator’s createSchemaTransformerV2
